client/my-sites/importer/section-import.jsx

```diff
@@ -130,98 +121,81 @@ class SectionImport extends Component {
 	} );
 
 	componentDidMount() {
-		ImporterStore.on( 'change', this.updateState );
 		this.updateFromAPI();
+		if ( this.props.isImporterStatusHydrated ) {
+			this.onceAutoStartImport();
+		}
 	}
 
 	componentDidUpdate() {
-		const { site } = this.props;
-
-		if ( ! site.ID ) {
-			return;
+		if ( this.props.isImporterStatusHydrated ) {
+			this.onceAutoStartImport();
 		}
 
-		this.onceAutoStartImport();
 		this.handleStateChanges();
 	}
 
-	componentWillUnmount() {
-		ImporterStore.off( 'change', this.updateState );
-	}
-
 	/**
 	 * Renders each enabled importer at the provided `state`
 	 *
-	 * @param {object} site Data for the currently active site
-	 * @param {string} siteTitle The site's title
-	 * @param {string} state The state constant for the importer components
+	 * @param {string} importerState The state constant for the importer components
 	 * @returns {Array} A list of react elements for each enabled importer
 	 */
-	renderIdleImporters( site, siteTitle, state ) {
-		const {
-			options: { is_wpcom_atomic: isAtomic },
-		} = site;
+	renderIdleImporters( importerState ) {
+		const { site, siteTitle } = this.props;
+		let importers = getImporters();
 
-		const importerElementsAll = getImporters();
-
-		/**
-		 * Filter out all importers except the WordPress ones for Atomic sites.
-		 */
-		const importerElementsFiltered = isAtomic
-			? importerElementsAll.filter( ( importer ) => importer.engine === 'wordpress' )
-			: importerElementsAll;
+		// Filter out all importers except the WordPress ones for Atomic sites.
+		if ( site.options.is_wpcom_atomic ) {
```

Comment:
We were previously destructuring those out from the `site` object. Any chance `site` or `site.options` would be `undefined` / `null`?